src/collider/BVHTree.cpp

```diff
@@ -322,91 +223,138 @@ uint32_t SingleBVHTree::Partition(SortKey *keys, uint32_t numKeys, const AABBd &
 	return startIdx;
 }
 
-void SingleBVHTree::ComputeOverlap(uint32_t nodeId, const AABBd &nodeAabb, std::vector<std::pair<uint32_t, uint32_t>> &out_isect) const
-{
-	PROFILE_SCOPED()
 
-	int32_t stackLevel = 0;
-	uint32_t *stack = stackalloc(uint32_t, m_treeHeight + 1);
-	// Push the root node
-	stack[stackLevel++] = 0;
+// ============================================================================
 
-	while (stackLevel > 0) {
-		const SingleBVHTree::Node *node = &m_nodes[stack[--stackLevel]];
 
-		if (!nodeAabb.Intersects(node->aabb))
-			continue;
+void BinnedAreaBVHTree::BuildNode(Node *node, SortKey *keys, uint32_t numKeys, const AABBd *objAabbs, uint32_t height)
```

Comment:
Two identical functions, maybe it’s still possible to do something about it?

But all I came up with is a template static method in a [friend base](https://github.com/pioneerspacesim/pioneer/commit/d8ae46ba557ff1cdc723e57f0aee495edf85a0b7)